tests/units/anta_tests/test_system.py

```diff
@@ -286,4 +287,92 @@
+                    },
+                    "3.3.3.3 (*.pool.ntp.org)": {
+                        "condition": "candidate",
+                        "peerIpAddr": "3.3.3.3",
+                    },
+                }
+            }
+        ],
+        "inputs": {"ntp_servers": [{"server_address": "1.1.1.1", "preferred": True}, {"server_address": "2.2.2.2"}, {"server_address": "3.3.3.3"}]},
+        "expected": {"result": "success"},
+    },
+    {
+        "name": "failure",
+        "test": VerifyNTPAssociations,
+        "eos_data": [
+            {
+                "peers": {
+                    "1.1.1.1 (*.pool.ntp.org)": {
+                        "condition": "candidate",
+                        "peerIpAddr": "1.1.1.1",
+                    },
+                    "2.2.2.2 (*.pool.ntp.org)": {
+                        "condition": "sys.peer",
+                        "peerIpAddr": "2.2.2.2",
+                    },
+                    "3.3.3.3 (*.pool.ntp.org)": {
+                        "condition": "candidate1",
+                        "peerIpAddr": "3.3.3.3",
+                    },
+                }
+            }
+        ],
+        "inputs": {"ntp_servers": [{"server_address": "1.1.1.1", "preferred": True}, {"server_address": "2.2.2.2"}, {"server_address": "3.3.3.3"}]},
+        "expected": {
+            "result": "failure",
+            "messages": [
+                "For NTP peer 1.1.1.1 expected condition as 'sys.peer' but found 'candidate' instead.\n"
+                "For NTP peer 2.2.2.2 expected condition as 'candidate' but found 'sys.peer' instead.\n"
+                "For NTP peer 3.3.3.3 expected condition as 'candidate' but found 'candidate1' instead.\n"
+            ],
+        },
+    },
+    {
+        "name": "failure-no-peers",
+        "test": VerifyNTPAssociations,
+        "eos_data": [{"peers": {}}],
+        "inputs": {"ntp_servers": [{"server_address": "1.1.1.1", "preferred": True}, {"server_address": "2.2.2.2"}, {"server_address": "3.3.3.3"}]},
+        "expected": {
+            "result": "failure",
+            "messages": ["NTP peers are not configured."],
+        },
+    },
+    {
+        "name": "failure-one-peer-no-found",
+        "test": VerifyNTPAssociations,
+        "eos_data": [
+            {
+                "peers": {
+                    "1.1.1.1 (*.pool.ntp.org)": {
+                        "condition": "sys.peer",
+                        "peerIpAddr": "1.1.1.1",
+                    },
+                    "2.2.2.2 (*.pool.ntp.org)": {
+                        "condition": "candidate",
+                        "peerIpAddr": "2.2.2.2",
+                    },
+                }
+            }
+        ],
+        "inputs": {"ntp_servers": [{"server_address": "1.1.1.1", "preferred": True}, {"server_address": "2.2.2.2"}, {"server_address": "3.3.3.3"}]},
+        "expected": {
+            "result": "failure",
+            "messages": ["NTP peer 3.3.3.3 is not configured."],
```

Comment:
add one more failure test having both ntp peer missing and condition as not expected failure message

Reply:
Handled